src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```
@@ -456,7 +456,40 @@ public static string GetPlatformApiLevel (string platformApiLevelName)
 
 		public static string GetLibraryImportDirectoryNameForAssembly (string assemblyIdentName)
 		{
-			return string.Concat (new MD2Managed ().ComputeHash (Encoding.UTF8.GetBytes (assemblyIdentName)).Take (5).Select (b => b.ToString ("X02")));
+			if (assemblyIdentityMap == null) {
+				throw new Exception ("Mapping was not loaded!");
```

You should throw `InvalidOperationException`, not `Exception`.